thinc/shims/pytorch.py

```diff
@@ -34,12 +39,20 @@ def __init__(
         optimizer: Any = None,
         mixed_precision: bool = False,
         grad_scaler: Optional[PyTorchGradScaler] = None,
+        device: Optional["torch.device"] = None,
     ):
         super().__init__(model, config, optimizer)
 
+        if device is None:
+            device = get_torch_default_device()
+        if model is not None:
```

Comment:
Is this check necessary? The rest of the code appears to have an implicit pre-condition that `model` is always a valid object (of the expected type).

Reply:
I had to add it because it occurred. IIRC it's when an existing transformer is deserialized. The shim is first constructed with `model=None` and then the model is put in place by deserialization.